emperor/core.py

```diff
@@ -194,15 +200,41 @@ def _repr_html_(self):
 
         return display(HTML(str(self)))
 
-    def _make_emperor(self):
-        """Private method to build an Emperor HTML string"""
-        output = []
+    def make_emperor(self, standalone=False):
+        """Build an emperor plot
+
+        Parameters
+        ----------
+        standalone : bool
+            Whether or not the produced plot should be a standalone HTML file.
+
+        Returns
+        -------
+        str
+            Formatted emperor plot.
+
 
-        with open(STYLE_PATH) as sty, open(MAIN_PATH) as mai:
-            style_template = Template(sty.read())
-            main_template = Template(mai.read())
+        Notes
+        -----
+        The `standalone` argument is intended for the different use-cases that
+        Emperor can have, either as an embedded widget that lives inside, for
+        example, the Jupyter notebook, or alternatively as an HTML file that
+        refers to resources locally. In this case you will probably also want
+        to copy the emperor_required_resources, and change the `base_url`
+        attribute of the object.
+        """
 
-        output.append(style_template.render(base_url=self.base_url))
+        # based on: http://stackoverflow.com/a/6196098
+        loader = FileSystemLoader(join(get_emperor_support_files_dir(),
+                                       'templates'))
+
+        if standalone:
+            main_path = basename(STANDALONE_PATH)
```

Comment:
Great point, I've added this!

On (Jun-28-16|10:40), Antonio Gonzalez wrote:
> >  
> > -        output.append(style_template.render(base_url=self.base_url))
> > +        # based on: http://stackoverflow.com/a/6196098
> > +        loader = FileSystemLoader(join(get_emperor_support_files_dir(),
> > +                                       'templates'))
> > +
> > +        if standalone:
> > +            main_path = basename(STANDALONE_PATH)
> 
> Should we also copy the resources files? If not, what's the solution for those occasions that we send a plot to a collaborator that has never used Emperor?
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/emperor/pull/510/files/378f456b29428865360dc7c3ad7a0beaed9e08b2#r68805359